### ISSUE TITLE
rule change: prefer-template should consider array.join()

### ISSUE THREAD

**Issue by Aladdin-ADD** [eslint/eslint#9070]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:**
* **npm Version:**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "prefer-template": "error"
}


```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
var foo = ["hello", "world"].join();
```

**What did you expect to happen?**

it should report an error.

**What actually happened? Please include the actual, raw output from ESLint.**

no error.

**Comment by Aladdin-ADD** [comment]
we need a way to enforce test cases:
```js
{
  code: [
    "foo",
    "bar"
  ].join("\n")
},

{
  code: unIndent`foo
    bar
  `
}
```

**Comment by platinumazure** [comment]
I could be okay with this as a new option, but I wouldn't want it on by default. Also, I would not want to run this rule on the codebase.

Also, the `unIndent` template tag function is something we defined in our codebase, and so would not be a general solution for rule violations.

At this point, I'm not convinced this is needed.

**Comment by platinumazure** [comment]
Also, if you could use this [rule change proposal template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md) in your first post, that would really help us out. Thanks!

**Comment by Aladdin-ADD** [comment]
so I didn't see a necessary to change this rule.
